### PR TITLE
[DSY-1420] feat: configure TextField to manually save its state

### LINKED ISSUE #102: Fix state handling when using more than one TextField in the same screen
**Describe the bug**
This is a common error when handling state for custom views with view groups. This error occurs when a we have a configuration change in a screen with more than one TextField. The value in the last TextField will be copied to the others TextFields, generating an unexpected behaviour. This error occurs because all EditTexts inside the TextFields on the same screen shares the same id.

**To Reproduce**
1. Go to NatDS Android Sample
2. Go to TextField Activity
3. Enable the last TextField and write some content in it
4. Force a configuration change on the device (Ex: Rotation)
5. See that all other TextFields get filled with an unexpected content

**Expected behavior**
The expected behaviour is to each TextField manage it's own behaviour. There shouldn't be any shared state that is not intentional for the user.

**Screenshots**

**Platform:**
 - OS: Android All Versions

**Additional context**
The same problem can be found here: https://stackoverflow.com/questions/39191312/android-compound-custom-view-restore-save-state

### PR DESCRIPTION
**Pull request by lucas-valadao**
# Description

This PR resolves #102. The `TextField` component is now configured to handle its state manually and individually (the implementation is following some instructions and code from [this link](https://www.netguru.com/codestories/how-to-correctly-save-the-state-of-a-custom-view-in-android). Also, there are some limitations related to state handling with views which don't have an id configured (this [link](https://medium.com/mobile-app-development-publication/a-must-know-secret-about-android-view-state-storing-mechanism-139d70487729) explains some of this problem, I couldn't find a better reference to it yet 😞 ); this PR also covers this scenario in its test cases.

Also:
- The footer icon was vertically configured as `center`. This modification also changes this configuration to `top`.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- The first test evaluates the behaviour of the `TextField` component when the activity gets recreated. It proves that a `TextField` component with an id configured is able to keep its state by keeping its `String` content.
- The second test evaluates the behaviour of the `TextField` component when the activity gets recreated. It proves that a `TextField` component without an id configured isn't able to keep its state since it didn't keep its `String` content.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: sample/src/test/kotlin/com/natura/android/sample/components/TextFieldActivityTest.kt [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+            onView(
+                allOf(
+                    hasDescendant(withInputType(InputType.TYPE_CLASS_NUMBER)),
+                    instanceOf(TextField::class.java)
+                )
+            ).check(hasValueInTextField(""))
+        }
+    }
+
+    private fun typeValueInTextField(string: String) = object : ViewAction {
```
Comment (mlcsouza) [inline sample/src/test/kotlin/com/natura/android/sample/components/TextFieldActivityTest.kt:61]: @lucas-valadao Esses eram os métodos da lib de teste commons? :D

File: sample/src/test/kotlin/com/natura/android/sample/components/TextFieldActivityTest.kt [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+            it.recreate()
+
+            onView(withId(R.id.text_item))
+                .check(hasValueInTextField(string))
+        }
+    }
+
+    @Test
+    fun `Should not save state after orientation change when component doesn't have id`() {
```
Comment (mlcsouza) [inline sample/src/test/kotlin/com/natura/android/sample/components/TextFieldActivityTest.kt:41]: Tô entendendo que o comportamento que teremos agora é:
Se textfield tem id, nós temos um controle certinho de estados. Caso ele não tenha, tbm não temos esse controle, entendi certinho?

Comment (lucas-valadao) [reply sample/src/test/kotlin/com/natura/android/sample/components/TextFieldActivityTest.kt:41]: Isso, correto!

File: sample/src/test/kotlin/com/natura/android/sample/components/TextFieldActivityTest.kt (hunk 1 above)
Comment (lucas-valadao) [reply sample/src/test/kotlin/com/natura/android/sample/components/TextFieldActivityTest.kt:61]: Sim, esses mesmo! 🙂

File: sample/src/test/kotlin/com/natura/android/sample/components/TextFieldActivityTest.kt (hunk 2 above)
Comment (lucas-valadao) [reply sample/src/test/kotlin/com/natura/android/sample/components/TextFieldActivityTest.kt:41]: Um ponto importante a considerar é que esse comportamento já acontecia no `TextField` antes, o que diferencia agora é que quando tem vários `TextField` na tela, o controle de estado não replica os estados considerando os ids iguais de diferentes custom views na tela; ele desabilita o controle de estado interno "automático" por id e passa a fazer isso considerando a árvore interna de views de cada custom view.

Comment (mlcsouza) [reply sample/src/test/kotlin/com/natura/android/sample/components/TextFieldActivityTest.kt:41]: Chique demais! Sucessos!

**Pull request merged**